xdsl/backend/riscv/register_allocation.py

```diff
@@ -15,7 +16,7 @@ class RegisterAllocator(abc.ABC):
     Base class for register allocation strategies.
     """
 
-    def __init__(self, limit_registers: int | None = None) -> None:
+    def __init__(self) -> None:
         pass
```

Comment:
Empty init can be dropped?

Reply:
Probably, I'll take a look

Reply:
I'm not sure exactly what the rules are for abstract base classes and inits. What I'd like to communicate here is that all subclasses should implement an init with no parameters, but I don't think this is actually enforced. I'll delete it for now until I have a better idea.